packages/caliper-samples/src/contract/scalardl/src/main/java/com/example/contract/smallbank/TransactSavings.java

```diff
@@ -0,0 +1,35 @@
+package com.example.contract.smallbank;
+
+import com.scalar.ledger.asset.Asset;
+import com.scalar.ledger.contract.Contract;
+import com.scalar.ledger.exception.ContractContextException;
+import com.scalar.ledger.ledger.Ledger;
+import java.util.Optional;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+
+public class TransactSavings extends Contract {
+
+  @Override
+  public JsonObject invoke(Ledger ledger, JsonObject argument, Optional<JsonObject> properties) {
+    String customerId = "" + argument.getInt(Const.KEY_CUSTOMER_ID);
+    int amount = argument.getInt(Const.KEY_AMOUNT);
+
+    Optional<Asset> asset = ledger.get(customerId);
+    if (!asset.isPresent()) {
+      throw new ContractContextException(Const.ERR_NOT_FOUND);
+    }
+
+    JsonObject data = asset.get().data();
+    int savingsBalance = data.getInt(Const.KEY_SAVINGS_BALANCE);
+    savingsBalance += amount;
```

Comment:
Just a question, this is called `TransactSavings` so this means that amount may be positive or negative?

Reply:
To be honest, I'm not exactly sure what they want to do but I imitated the contract in HL Fabric. In the original contract in Fabric has some comments like this.

```
func (t *SmallbankChaincode) TransactSavings(stub shim.ChaincodeStubInterface, args []string) pb.Response {
	if len(args) != 2 {	// should be [amount,customer_id]
		return errormsg(ERROR_WRONG_ARGS + " transaction_savings")
	}
	account, err := loadAccount(stub, args[1])
	if err != nil {
		return errormsg(ERR_NOT_FOUND);
	}
	amount, _ := strconv.Atoi(args[0])
	// since the contract is only used for perfomance testing, we ignore this check 
	//if amount < 0 && account.SavingsBalance < (-amount) {
	//	return errormsg("Insufficient funds in source savings account")
	//}
	account.SavingsBalance += amount
	err = saveAccount(stub, account)
	if err != nil {
		return systemerror(err.Error())
	}

	return shim.Success(nil)
}
```

Same as Fabric, I did not care error checking.

Reply:
Haha. I see. Sounds good to me.